### PR TITLE
Separate XML and HTML renderers.

### PR DESCRIPTION
**Pull request by mbrandonw**
This creates a separate rendering function for rendering vanilla XML documents that don't need any of the "void element" logic.

### CODE REVIEW

File: Sources/Html/XmlRender.swift [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/// Renders an array of nodes to an XML string.
+///
+/// - Parameter nodes: An array of nodes.
+public func xmlRender(_ nodes: [Node]) -> String {
```
Comment (stephencelis) [inline Sources/Html/XmlRender.swift]: What do you think of keeping this "private" for now while we consider some alternatives?
```suggestion
public func _xmlRender(_ nodes: [Node]) -> String {
```
I was thinking we could introduce some kind of helper strategy functions that handle this stuff...
``` swift
render(nodes, as: .html5)
// ...or...
render(nodes, as: .xml)
```

File: Sources/Html/XmlRender.swift [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  return nodes.reduce(into: "") { $0.append(xmlRender($1)) }
+}
+
+/// Renders a node to an XML string.
+///
+/// - Parameter node: A node.
+public func xmlRender(_ node: Node) -> String {
```
Comment (stephencelis) [inline Sources/Html/XmlRender.swift]: (This too if you agree.)
```suggestion
public func _xmlRender(_ node: Node) -> String {
```

File: Sources/Html/XmlRender.swift (hunk 1 above)
Comment (mbrandonw) [reply Sources/Html/XmlRender.swift]: ah yeah I'm down to do that so that we can explore more API's of this soon. will do!

Comment (mbrandonw) [reply Sources/Html/XmlRender.swift]: ah yeah I'm down to do that so that we can explore more API's of this soon. will do!

File: .gitignore [hunk 3]
```diff
@@ -70,3 +70,4 @@ fastlane/test_output
 .DS_Store
 Packages/*
 Package.resolved
+.swiftpm
```
Comment (stephencelis) [inline .gitignore:73]: Is there a "best practice" on this yet? Like should some files within not be ignored?

Comment (stephencelis) [reply .gitignore:73]: Ah: https://forums.swift.org/t/files-created-when-opening-spm-package-in-xcode-11/28589

Comment (stephencelis) [reply .gitignore:73]: Also: https://github.com/github/gitignore/blob/master/Swift.gitignore#L45-L47

**Review by stephencelis — APPROVED** [review]

File: .gitignore (hunk 3 above)
Comment (mbrandonw) [reply .gitignore:73]: It only seems to contain a workspace and some user data, so I was thinking it should always be ignored. I think I've done it on other repos too 🤔

But I haven't yet read about doing one way or the other yet...

**Pull request merged**